File: leetcode36.py
class Solution:
    def isValidSudoku(self, board: list[list[str]]) -> bool:
        columnTracker = {
            0: [],
            1: [],
            2: [],
            3: [],
            4: [],
            5: [],
            6: [],
            7: [],
            8: []
        }
        subBoxTracker = {
            '00': [],
            '03': [],
            '06': [],
            '30': [],
            '33': [],
            '36': [],
            '60': [],
            '63': [],
            '66': []
        }
        rowTracker = []

        #TODO Switch x and y variables around for easier understanding
        for x in range(9):
            for y in range(9):
                if board[x][y] in rowTracker:
                    return False
                elif board[x][y] in columnTracker[y]:
                    return False
                else:
                    if board[x][y] != ".":
                        rowTracker.append(board[x][y])
                        columnTracker[y].append(board[x][y])
            rowTracker.clear()

        for i in (0,3,6):
            for j in (0,3,6):
                box = str(i) + str(j)
                for l in range(i, i+3):
                    for k in range(j, j+3):
                        if board[l][k] in subBoxTracker[box]:
                            return False
                        else:
                            if board[l][k] != ".":
                                subBoxTracker[box].append(board[l][k])
        return True

Solution.isValidSudoku(Solution,[
    [".",".",".",".","5",".",".","1","."],
    [".","4",".","3",".",".",".",".","."],
    [".",".",".",".",".","3",".",".","1"],
    ["8",".",".",".",".",".",".","2","."],
    [".",".","2",".","7",".",".",".","."],
    [".","1","5",".",".",".",".",".","."],
    [".",".",".",".",".","2",".",".","."],
    [".","2",".","9",".",".",".",".","."],
    [".",".","4",".",".",".",".",".","."]])

File: javaStuff/leetcode36.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        HashMap<Integer, ArrayList<Character>> columnTracker = new HashMap<>();
        columnTracker.put(0, new ArrayList<Character>());
        columnTracker.put(1, new ArrayList<Character>());
        columnTracker.put(2, new ArrayList<Character>());
        columnTracker.put(3, new ArrayList<Character>());
        columnTracker.put(4, new ArrayList<Character>());
        columnTracker.put(5, new ArrayList<Character>());
        columnTracker.put(6, new ArrayList<Character>());
        columnTracker.put(7, new ArrayList<Character>());
        columnTracker.put(8, new ArrayList<Character>());
        
        HashMap<String, ArrayList<Character>> subBoxTracker = new HashMap<>();
        subBoxTracker.put("00", new ArrayList<Character>());
        subBoxTracker.put("03", new ArrayList<Character>());
        subBoxTracker.put("06", new ArrayList<Character>());
        subBoxTracker.put("30", new ArrayList<Character>());
        subBoxTracker.put("33", new ArrayList<Character>());
        subBoxTracker.put("36", new ArrayList<Character>());
        subBoxTracker.put("60", new ArrayList<Character>());
        subBoxTracker.put("63", new ArrayList<Character>());
        subBoxTracker.put("66", new ArrayList<Character>());
        
        ArrayList<Character> rowTracker = new ArrayList<>();
        
        for(int i = 0; i < 9; i++){
            for(int j = 0; j < 9; j++){
                if(rowTracker.contains(board[i][j])) return false;
                else if(columnTracker.get(j).contains(board[i][j])) return false;
                else{
                    if(board[i][j] != '.'){
                        rowTracker.add(board[i][j]);
                        columnTracker.get(j).add(board[i][j]);
                    }
                }
            }
            rowTracker.clear();
        }
        
        for(int i = 0; i < 9; i += 3){
            for(int j = 0; j < 9; j += 3){
                String box = Integer.toString(i) + Integer.toString(j);
                for(int l = i; l < i+3; l++){
                    for(int k = j; k < j+3; k++){
                        if(subBoxTracker.get(box).contains(board[l][k])) return false;
                        else{
                            if(board[l][k] != '.') subBoxTracker.get(box).add(board[l][k]);
                        }
                    }
                }
            }
        }
        
        return true;
    }
}